elasticdl/python/master/servicer.py

```diff
@@ -339,13 +342,13 @@ def ReportGradient(self, request, _):
                             self._model[name].numpy().shape[0] - 1,
                         )
                     )
-                indexed_grads[name] = arr
+                indexed_grads[name] = tensor.to_tf_tensor()
             else:
-                if arr.shape != self._model[name].numpy().shape:
+                if tensor.values.shape != self._model[name].numpy().shape:
                     raise ValueError(
                         "Gradient key: %s has incompatible dimension", name
                     )
-                tmp[name] = arr
+                tmp[name] = tensor.to_tf_tensor()
```

Comment:
Prefer use an explicit name for tmp

Reply:
Yes. We can rename it in another PR and make this PR more clean.